abci/tests/grpc.rs

```diff
@@ -0,0 +1,149 @@
+//! Test gRPC server for ABCI protocol.
+//!
+//! This test verifies that the gRPC server generated with tonic as part of the
+//! tenderdash-proto crate can successfully connect to Tenderdash instance.
+//!
+//! This test should be implemented in the tenderdash-proto crate; however, it
+//! is implemented here to use already existing docker container testing
+//! logic.
+#![cfg(feature = "grpc")]
```

Comment:
Is this for Arc?

Reply:
This is for whole file.

> Inner attributes, written with a bang (!) after the hash (#), apply to the item that the attribute is declared within. Outer attributes, written without the bang after the hash, apply to the thing that follows the attribute.

https://doc.rust-lang.org/reference/attributes.html